lib/cockpit-components-tooltip.jsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+"use strict";
+
+var React = require('react');
+
+/* A tooltip, styled via Patternfly/Bootstrap.
+ *
+ * Example:
+ *
+ *   <Tooltip tip="Insufficient privileges">
+ *      <button disabled>Reboot</button>
+ *   </ToolTip>
+ *
+ * Whenever the mouse hovers over the children of a Tooltip,
+ * the text (or arbitrary element) in the "tip" property is shown.
+ */
+
+var Tooltip = React.createClass({
+    getInitialState: function () {
+        return { open: false };
+    },
+    onMouseover: function () {
+        this.setState({ open: true });
+    },
+    onMouseout: function () {
+        this.setState({ open: false });
+    },
+    render: function () {
+        var self = this;
+
+        if (!self.props.tip)
+            return self.props.children;
+
+        // Placement
```

Comment:
It's a personal preference, I guess.  Also, Emacs web-mode is shitty with `/* */` comments.